### PR TITLE
Jsonform component

### PR DESCRIPTION
**Pull request by tanmaythole**
Example Schema:

> [{
      label: "Email",
      type: "text",
      name: "email",
      required: true,
      placeholder: "Enter email",
    },
    {
      label: "Password",
      type: "password",
      name: "password",
      required: true,
      placeholder: "Enter password",
    },
    {
      wrapperClass: "mb-3 w-100 text-end fs-6 fst-italic",
      component: `<Link to={FORGOT_PASSWORD_ROUTE}>Forgot Password?</Link>`,
    },
    {
      component: `(<Button  type="submit" iconType="save" className="w-100" isLoading={isLoading}>Login</Button>)`,
    }]

### CODE REVIEW

File: src/Components/JSONForm/utils/getSchemaField.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Controller } from "react-hook-form";
+
+import Input from "Components/forms/Inputs/Input";
+
+const getSchemaField = (name, type, control, errors, props) => {
```
Comment (tanmaythole) [inline src/Components/JSONForm/utils/getSchemaField.js]: set default value of `errors`

Comment (tanmaythole) [reply src/Components/JSONForm/utils/getSchemaField.js]: 👍🏻

**Comment by ChinmaySonawane** [comment]
What about validation and default value?

**Comment by ChinmaySonawane** [comment]
We can also consider checkbox, radio buttons, switch, select, async select, datepicker for this schema type

**Comment by ChinmaySonawane** [comment]
Array logic is missing?

File: src/Components/JSONForm/index.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      {schema?.map(
+        (
+          { name, wrapper, wrapperClass, wrapperProps, type, ...properties },
```
Comment (ChinmaySonawane) [inline src/Components/JSONForm/index.jsx]: no need to destructor here we can destructor while passing this to SchemaField

File: src/Components/JSONForm/utils/getSchemaField.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { Controller } from "react-hook-form";
+
+import Input from "Components/forms/Inputs/Input";
+
+const getSchemaField = (name, type, control, errors = {}, props) => {
```
Comment (ChinmaySonawane) [inline src/Components/JSONForm/utils/getSchemaField.js]: propTypes are missing for this component

Comment (ChinmaySonawane) [reply src/Components/JSONForm/utils/getSchemaField.js]: This is not a normal function this should be component

Comment (tanmaythole) [reply src/Components/JSONForm/utils/getSchemaField.js]: Okay

Comment (tanmaythole) [reply src/Components/JSONForm/utils/getSchemaField.js]: Done 👍🏻

File: src/Components/JSONForm/index.jsx (hunk 2 above)
Comment (tanmaythole) [reply src/Components/JSONForm/index.jsx]: Done 👍🏻

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**